### PR TITLE
[nrf fromtree] Bluetooth: Host: Support for PAwR

### PR DESCRIPTION
**Pull request by hermabe**
Cherry-picks host support for Periodic Advertising with Responses (PAwR).

### CODE REVIEW

**Review by alwa-nordic — APPROVED** [review]

**Comment by hermabe** [comment]
- Rebased after upmerge.
- Added one noup commit that cherry-picks parts of an upstream commit. To be dropped in next upmerge, not sure if any other sauce-tag fits better.

**Comment by hermabe** [comment]
Two CI issues, both unrelated to the changes in this PR:
- "Pull Request Assigner" failure is a known issue
- Jenkins failure is because of incompatibilities between sdk-mcuboot and upstream mcuboot.

**Review by vegu-nordic — APPROVED** [review]

File: samples/bluetooth/periodic_adv_rsp/prj.conf [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+CONFIG_BT_MAX_CONN=1
+CONFIG_BT_CENTRAL=y
+CONFIG_BT_PER_ADV_SYNC_TRANSFER_SENDER=y
+
+CONFIG_BT_PER_ADV_RSP=y
+
+CONFIG_BT_REMOTE_INFO=y
+CONFIG_BT_GATT_CLIENT=y
+
+CONFIG_LOG=y
```
Comment (pirun) [inline samples/bluetooth/periodic_adv_rsp/prj.conf:16]: Missing CONFIG_BT_CTLR_SDC_PAWR_ADV=y

File: samples/bluetooth/periodic_adv_conn/src/main.c [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	.num_subevents = NUM_SUBEVENTS,
+	.subevent_interval = SUBEVENT_INTERVAL,
+	.response_slot_delay = 0x5,
+	.response_slot_spacing = 0x50,
+	.num_response_slots = NUM_RSP_SLOTS,
+};
+
+static struct bt_le_per_adv_subevent_data_params subevent_data_params[NUM_SUBEVENTS];
+static struct net_buf_simple bufs[NUM_SUBEVENTS];
+static uint8_t backing_store[PACKET_SIZE][NUM_SUBEVENTS];
```
Comment (pirun) [inline samples/bluetooth/periodic_adv_conn/src/main.c:29]: Should this be static uint8_t backing_store[NUM_SUBEVENTS][PACKET_SIZE]; ?

File: samples/bluetooth/periodic_sync_rsp/prj.conf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+CONFIG_BT_MAX_CONN=1
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_PER_ADV_SYNC_TRANSFER_SENDER=y
+CONFIG_BT_PER_ADV_SYNC_TRANSFER_RECEIVER=y
+
+CONFIG_BT_PER_ADV_SYNC_RSP=y
+CONFIG_BT_PER_ADV_SYNC_BUF_SIZE=247
+
+CONFIG_LOG=y
```
Comment (pirun) [inline samples/bluetooth/periodic_sync_rsp/prj.conf:16]: Missing CONFIG_BT_CTLR_SDC_PAWR_SYNC=y

**Comment by hermabe** [comment]
@pirun The softdevice controller symbols are defaulted in sdk-nrf. We do not want to reference anything sdc-specific in zephyr and any changes here have to be done upstream first.

**Comment by hermabe** [comment]
Rebased on main and added a commit for fixing phy translation.

File: samples/bluetooth/periodic_adv_conn/src/main.c (hunk 2 above)
Comment (alwa-nordic) [reply samples/bluetooth/periodic_adv_conn/src/main.c:29]: This should be fixed upstream. Will you post a PR, @hermabe?

**Review by alwa-nordic — CHANGES_REQUESTED** [review]
A problem was found with an array declaration. https://github.com/nrfconnect/sdk-zephyr/pull/1138#discussion_r1185718288

This will result in malfunction, since it will affect logic where `ARRAY_SIZE` is used. `init_bufs` seems to assume that `ARRAY_SIZE(backing_store) == ARRAY_SIZE(bufs)`

File: samples/bluetooth/periodic_adv_conn/src/main.c (hunk 2 above)
Comment (hermabe) [reply samples/bluetooth/periodic_adv_conn/src/main.c:29]: https://github.com/zephyrproject-rtos/zephyr/pull/57591

**Pull request merged**